Tests/SonarScanner.MSBuild.Common.Test/MsBuildPathSettingsTests.cs

```diff
@@ -34,10 +34,10 @@ public void GetImportBeforePaths_NonWindows_AppData_Is_NullOrEmpty()
         {
             Action action;
 
-            action = new Action(() => new MsBuildPathSettings((folder, o) => string.Empty, IsWindows(false), DirExists(true)).GetImportBeforePaths());
+            action = new Action(() => MsBuildPathSettings(string.Empty, OS.Linux, _ => true).GetImportBeforePaths());
             action.Should().ThrowExactly<IOException>().WithMessage("Cannot find local application data directory.");
 
-            action = new Action(() => new MsBuildPathSettings((folder, o) => null, IsWindows(false), DirExists(true)).GetImportBeforePaths());
+            action = new Action(() => MsBuildPathSettings(null as string, OS.Linux, _ => true).GetImportBeforePaths());
```

Comment:
I would prefer a named argument for overload disambiguation here (if the parameters are named differently) instead of `null as string`.